SCF checkpointing update (#2190)
psi4/psi4: doc/sphinxman/source/scf.rst

    energy('scf')

.. .. _`scf:restart`:

Restarting the SCF
~~~~~~~~~~~~~~~~~~

Reading orbital data from a previous calculations is done via the ``restart_file`` option,
where the actual file is a serialized ``wfn`` object (see :ref:`saving the wfn <sec:save_wfn>` )

> By default the orbital data file of the converged SCF(``psi.PID.name.180.npy``) is deleted after |PSIfour| exists or the ``clean()`` function is called.

Reviewer: Line is too long. Also `deleted after |PSIfour| exits` not `exists`.

To keep the orbital data and also write the orbitals after every iteration the ``write_orbitals`` options is available: ::

  energy('scf', write_orbitals='my_mos'),

which write a ``Wavefunction`` object converted (serialized) to a numpy file called ``my_mos.npy``.
The restart can then be done as follows: ::

  set guess read
  energy('scf', restart_file='my_mos')

> The ``.npy`` ending is optional.

Reviewer: Specifying the ``.npy`` suffix when writing and reading restart files is optional.

    energy('scf')

.. .. _`scf:restart`:

Restarting the SCF
~~~~~~~~~~~~~~~~~~

Reading orbital data from a previous calculations is done via the ``restart_file`` option,
where the actual file is a serialized ``wfn`` object (see :ref:`saving the wfn <sec:save_wfn>`)

> By default the orbital data file of the converged SCF(``psi.PID.name.180.npy``) is deleted

Reviewer 2: ```suggestion
By default, the orbital data file of the converged SCF(``psi.PID.name.180.npy``) is deleted
```

after |PSIfour| exits or the ``clean()`` function is called. The orbital guess is automatically
set to ``READ`` when ``restart_file`` is given a ``wfn`` file.
To write the orbitals after every iteration and keep the orbitals from the last iteration, the ``write_orbitals`` options is available: ::

  energy('scf', write_orbitals='my_mos'),

which writes a ``Wavefunction`` object converted (serialized) to a numpy file called ``my_mos.npy``.
The restart can then be done as follows: ::

  energy('scf', restart_file='my_mos')

Specifying the ``.npy`` suffix when writing and reading restart files is optional.

Alternatively, the restart can also be done from any previously saved ``wfn`` object. ::

  energy, scf_wfn = energy('scf',return_wfn=True)
  scf_wfn.to_file('my_wfn')
  energy('scf', restart_file='my_wfn')

For advanced users manipulating or writing custom wavefunction files, note

> that |PSIfour| expects the ``.npy`` suffix for the numpy file on disk (e.g. not `.npz`).

Reviewer 2: ```suggestion
that |PSIfour| expects the numpy file on disk to have the ``.npy`` extension, not, e.g., `.npz`.
```

    energy('scf')

.. .. _`scf:restart`:

Restarting the SCF
~~~~~~~~~~~~~~~~~~

Reading orbital data from a previous calculations is done via the ``restart_file`` option,
where the actual file is a serialized ``wfn`` object (see :ref:`saving the wfn <sec:save_wfn>`)
By default the orbital data file of the converged SCF(``psi.PID.name.180.npy``) is deleted
after |PSIfour| exits or the ``clean()`` function is called.

> To keep the orbital data and also write the orbitals after every iteration the ``write_orbitals``

Reviewer 2: ```suggestion
To write the orbitals after every iteration and keep the orbitals from the last iteration, the ``write_orbitals``
```

options is available: ::

  energy('scf', write_orbitals='my_mos'),

> which write a ``Wavefunction`` object converted (serialized) to a numpy file called ``my_mos.npy``.

Reviewer 2: ```suggestion
which writes a ``Wavefunction`` object converted (serialized) to a numpy file called ``my_mos.npy``.
```

The restart can then be done as follows: ::

  set guess read
  energy('scf', restart_file='my_mos')

Specifying the ``.npy`` suffix when writing and reading restart files is optional.

Alternatively, the restart can also be done from any previously saved ``wfn`` object. ::

  energy, scf_wfn = energy('scf',return_wfn=True)
  scf_wfn.to_file('my_wfn')
  set guess read
  energy('scf', restart_file='my_wfn')

> Setting the orbital guess explicitly to ``read`` is crucial as this is not done automatically when 

Reviewer 2: Why not? Is there ever a case where a user is going to supply this argument and _not_ want to read the orbitals?

Author: Not sure. I stayed with the old behaviour initially but I think I would also favour automatically turning on `READ`

Author: Next commit changes the behaviour to automatically enable `guess read` for the local scope.

a ``restart_file`` is specified.

> For advanced users manipulating or writing custom wavefunction files, note that |PSIfour| expects

Reviewer 2: Can you give more context? I don't see Psi needing the `.npy` suffix in your example above.

Author: psi4 does not need npy in the python arguments, but present on the disk.
It won't read e.g. `.npz` files or whatever users come up with.
